implementations/knockout/app/topics/topics.js

```diff
@@ -0,0 +1,127 @@
+var model = require('model.js');
+var ko = require('knockout');
+
+
+module.exports = function(stateRouter) {
+	stateRouter.addState({
+		name: 'app.topics',
+		route: '/topics',
+		defaultChild: 'no-task',
+		template: require('fs').readFileSync('implementations/knockout/app/topics/topics.html', 'utf8'),
+		resolve: function(data, parameters, cb) {
+			var viewModel = new TopicsVM(stateRouter);
+			cb(null, viewModel);
+		},
+		activate: function(context) {
+			var viewModel = context.content;
+			viewModel.activate();
+			context.on('destroy', function() {
+				viewModel.dispose();
+			});
+		}
+	});
+
+	require('./tasks/tasks')(stateRouter);
+};
+
+
+function TopicsVM(stateRouter) {
+	this._stateRouter = stateRouter;
+	this.makePath = this._stateRouter.makePath.bind(this._stateRouter);
+
+	this.addingTopic = ko.observable(false);
+	this.newTopic = ko.observable('');
+	this.topics = ko.observableArray();
+	this.activeStateTracker = ko.observable();
+
+	this.onTasksSaved = TopicsVM.prototype.onTasksSaved.bind(this);
+	this.onTopicsSaved = TopicsVM.prototype.onTopicsSaved.bind(this);
+	this.onStateChangeEnd = TopicsVM.prototype.onStateChangeEnd.bind(this);
+	this.stateIsActive = TopicsVM.prototype.stateIsActive.bind(this);
+
+	model.on('tasks saved', this.onTasksSaved);
+	model.on('topics saved', this.onTopicsSaved);
+	this._stateRouter.on('stateChangeEnd', this.onStateChangeEnd);
+}
+
+ko.utils.extend(TopicsVM.prototype, {
+	activate: function() {
+		this.addingTopic(false);
+		this.newTopic('');
+		this._init();
+	},
+
+	dispose: function() {
+		model.removeListener('tasks saved', this.onTasksSaved);
+		model.removeListener('topics saved', this.onTopicsSaved);
+		this._stateRouter.removeListener('stateChangeEnd', this.onStateChangeEnd);
+	},
+
+	addTopic: function() {
+		var newTopicName = this.newTopic();
+
+		if (this.addingTopic()) {
+			if (newTopicName) {
+				var newTopic = model.addTopic(newTopicName);
+				model.saveTopics();
+				this.newTopic('');
+				this.addingTopic(false);
+
+				this._stateRouter.go('app.topics.tasks', {
+					topicId: newTopic.id
+				});
+			}
+		}
+		else {
+			this.addingTopic(true);
+		}
+	},
+
+	onTasksSaved: function(topicId) {
+		this._recomputeTasksLeft(topicId);
+	},
+
+	onTopicsSaved: function() {
+		this._init();
+	},
+
+	onStateChangeEnd: function() {
+		this.activeStateTracker.valueHasMutated();
```

Comment:
I think this should be done at the renderer level, instead of each state having to be responsible for updating the active state after it changes.  Something global, like what the Riot renderer does: https://github.com/TehShrike/riot-state-renderer/blob/master/index.js#L12-L14